qgreenland/util/model_validators/layer_style.py

```diff
@@ -0,0 +1,160 @@
+        continuous_legend_suffix = (
+            rampLegendSettings.attrib["suffix"] if rampLegendSettings else None
+        )
+
+        if continuous_legend_suffix == unit_suffix:
+            if rampLegendSettings and rampLegendSettings.attrib["orientation"] != "1":
+                raise exc.QgrInvalidConfigError(
+                    f"{error_prefix}\n"
+                    "    Continuous legend orientation must be horizontal.\n"
+                    f"In QGIS >=3.28, edit this style ({style_filepath}) in"
+                    " the layer symbology menu and set the orientation in the"
+                    ' "Legend Settings" menu.'
+                )
+
+            # This style's suffixes and orientation are set up correctly!
+            return style_name
+
+        # QGIS can be really finicky with the "Label unit suffix" field because its
+        # value is not saved into the XML as a dedicated field like one would expect.
+        # Instead, the labels are pre-calculated to add prefixes and suffixes, then
+        # saved in whole to the XML. This means that for QGIS to display a "Label unit
+        # suffix" field in the GUI, it has to calculate the suffixes and prefixes from
+        # the labels in the XML. Strange errors can be introduced during this process,
+        # like seemingly-random numbers being prepended to the "Label unit suffix" value
+        # in the GUI, and then propagating to all labels in the XML.
+        raise exc.QgrInvalidConfigError(
+            f"{error_prefix}\n"
+            f"    {continuous_legend_suffix=} != {unit_suffix=}\n"
+            f"In QGIS >=3.28, edit this style ({style_filepath}) in the layer symbology"
+            ' menu and ensure that the "Label unit suffix" exactly matches the "suffix"'
+            ' field in the "Legend Settings" menu. If the suffixes reported in this'
+            ' error message look wrong to you, you may need to re-check the "Label unit'
+            ' suffix" and press the "Classify"  button and re-save the style to correct'
+            " errors."
+        )
+
+
+def _get_unit_suffix(*, label: str, value: str, label_precision: int = 0) -> str | None:
+    """Calculate the unit suffix for a QGIS colormap entry.
+
+    A QGIS style's `<colorrampshader>` has a `labelPrecision` attribute. Each `<item>`
+    within has a `label` and a `value` attribute. In order to calculate the suffix from
+    the `label`, we need to calculate the non-suffix part of the `label` based on
+    `value`.  Then we can difference the calculated label (sans suffix) and the real
+    label to get the suffix.
+
+    TODO: Support a "clip" argument (originates in `<colorrampshader>` element)? Not
+    sure how that's set in the GUI but seems to usually be 0.
+
+    TODO: Does not support large numbers well, adds commas where the QGIS symbology menu
+    does not.
+            formatter.formatNumber(float(-9902))
+            >>> '-9,902'
```

Comment:
I'm a bit frustrated running in to this after I thought I had things figured out. Is there any way to re-use  QGIS' behavior for calculating these labels? What am I doing wrong?

Reply:
We decided to just check that a suffix was populated for linearly-interpolated colorramps, not actually check for consistency with item labels.